File: CS/WinSolution.Module.Win/ProductSearchObject.cs
using System;
using DevExpress.Xpo;
using Dennis.Search.Win;
using DevExpress.Persistent.Base;

namespace WinSolution.Module.Win {
    [NonPersistent]
    [System.ComponentModel.DisplayName("Complex search by products")]
    [ImageName("Attention")]
    public class ProductSearchObject : SearchObjectBase<Product> {
        public ProductSearchObject(Session session) : base(session) { }
        public override void Reset() {
            Price = null;
            Name = null;
            //Dennis: it's very important to reset searched properties before the base method call.
            base.Reset();
        }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
        private decimal? _Price;
        //Dennis: it's also recommended to use nullable properties here.
        public decimal? Price {
            get { return _Price; }
            set { SetPropertyValue("Price", ref _Price, value); }
        }
    }
}

File: VB/WinSolution.Module.Win/ProductSearchObject.vb
Imports DevExpress.Xpo
Imports Dennis.Search.Win
Imports DevExpress.Persistent.Base

Namespace WinSolution.Module.Win

    <NonPersistent>
    <System.ComponentModel.DisplayName("Complex search by products")>
    <ImageName("Attention")>
    Public Class ProductSearchObject
        Inherits SearchObjectBase(Of Product)

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Overrides Sub Reset()
            Price = Nothing
            Name = Nothing
            'Dennis: it's very important to reset searched properties before the base method call.
            MyBase.Reset()
        End Sub

        Private _Name As String

        Public Property Name As String
            Get
                Return _Name
            End Get

            Set(ByVal value As String)
                SetPropertyValue("Name", _Name, value)
            End Set
        End Property

        Private _Price As Decimal?

        'Dennis: it's also recommended to use nullable properties here.
        Public Property Price As Decimal?
            Get
                Return _Price
            End Get

            Set(ByVal value As Decimal?)
                SetPropertyValue("Price", _Price, value)
            End Set
        End Property
    End Class
End Namespace
